pills/08/autotools-nix.txt

```diff
@@ -1,11 +1,10 @@
 pkgs: attrs:
-  with pkgs;
   let defaultAttrs = {
-    builder = "${bash}/bin/bash";
+    builder = "${pkgs.bash}/bin/bash";
     args = [ ./builder.sh ];
-    baseInputs = [ gnutar gzip gnumake gcc coreutils gawk gnused gnugrep binutils.bintools ];
+    baseInputs = with pkgs; [ gnutar gzip gnumake gcc coreutils gawk gnused gnugrep binutils.bintools ];
     buildInputs = [];
     system = builtins.currentSystem;
   };
   in
-  derivation (defaultAttrs // attrs)
+  pkgs.stdenv.mkDerivation (defaultAttrs // attrs)
```

Comment:
I'm pretty sure the purpose of the exercise was developing `mkDerivation` from scratch, therefore we only use builtins. I'm not familiar with the exact text here, but I assume that `pkgs` are just derivations and not stuff specifically from Nixpkgs.

Reply:
`pkgs` do come from Nixpkgs:

https://github.com/NixOS/nix-pills/blob/29e1e273b827db169359256f11bf801499c89e4a/pills/08/hello-nix-rev-2.txt#L3

and making `with` more localized makes sense with how [preferences evolved](https://nix.dev/recipes/best-practices#with-scopes).

But as you say, `pkgs.stdenv.mkDerivation` definitely should not be used.

Reply:
Hi both, thanks a lot for the feedback.

I have changed to using derivation instead of pkgs.stdenv.mkDerivation now, and also changed the name to mkDerivation as there is no confusion or collision about the naming now.